Source/WebKit/UIProcess/mac/WebPageProxyMac.mm

```diff
@@ -182,12 +182,20 @@ static inline bool expectsLegacyImplicitRubberBandControl()
 
 void WebPageProxy::windowAndViewFramesChanged(const FloatRect& viewFrameInWindowCoordinates, const FloatPoint& accessibilityViewCoordinates)
 {
-    if (!hasRunningProcess())
-        return;
-
     // In case the UI client overrides getWindowFrame(), we call it here to make sure we send the appropriate window frame.
     m_uiClient->windowFrame(*this, [this, protectedThis = Ref { *this }, viewFrameInWindowCoordinates, accessibilityViewCoordinates] (FloatRect windowFrameInScreenCoordinates) {
         FloatRect windowFrameInUnflippedScreenCoordinates = pageClient().convertToUserSpace(windowFrameInScreenCoordinates);
+
+        if (!hasRunningProcess()) {
```

Comment:
There's some statefulness here that is tricky. We assign to `m_viewWindowCoordinates` here, but never clear it. Can we just always do this caching, and always fill `parameters.viewWindowCoordinates` on process creation? I would prefer a stateless solution.

Reply:
As we discussed in the person, this is tricky to do because m_uiClient->windowFrame returns the window frame asynchronously. Will land as is. When web content process get relaunched, it would use the newly cached value, and we update the cached value whenever window moves / resizes so we should be good here.